src/edit_view.rs

```diff
@@ -109,153 +195,136 @@ impl EditView {
 
-        // let overlay: Overlay = frame_builder.get_object("overlay").unwrap();
-        // let search_revealer: Revealer = frame_builder.get_object("revealer").unwrap();
-        // let frame: Frame = frame_builder.get_object("frame").unwrap();
-        let search_entry: SearchEntry = find_rep_builder.get_object("search_entry").unwrap();
-        let go_down_button: Button = find_rep_builder.get_object("go_down_button").unwrap();
-        let go_up_button: Button = find_rep_builder.get_object("go_up_button").unwrap();
+        let edit_view = Rc::new(RefCell::new(EditView {
+            core: core.clone(),
+            main_state: main_state.clone(),
+            file_name,
+            pristine: true,
+            view_id: view_id.to_string(),
+            root_widget: EditView::get_root_box(&view_item, &find_replace),
+            top_bar: TopBar::new(),
+            view_item: view_item.clone(),
+            line_cache: LineCache::new(),
+            edit_font: EditView::get_edit_font(&pango_ctx, &main_state.borrow().config),
+            interface_font: EditView::get_interface_font(&pango_ctx),
+            find_replace: find_replace.clone(),
+        }));
 
-        let root_box = Box::new(Orientation::Vertical, 0);
-        let hbox = Box::new(Orientation::Horizontal, 0);
-        let vbox = Box::new(Orientation::Vertical, 0);
-        root_box.pack_start(&search_bar, false, false, 0);
-        root_box.pack_start(&hbox, true, true, 0);
-        hbox.pack_start(&linecount_da, false, false, 0);
-        hbox.pack_start(&vbox, true, true, 0);
-        hbox.pack_start(&vscrollbar, false, false, 0);
-        vbox.pack_start(&da, true, true, 0);
-        vbox.pack_start(&hscrollbar, false, false, 0);
-        root_box.show_all();
+        edit_view.borrow_mut().update_title();
+        crate::MainWin::set_language(&core, view_id, "Plain Text");
 
-        // Make the widgets for the tab
-        let tab_hbox = gtk::Box::new(Orientation::Horizontal, 5);
-        let label = Label::new(Some(""));
-        tab_hbox.add(&label);
-        let close_button = Button::new_from_icon_name("window-close", IconSize::Button);
-        tab_hbox.add(&close_button);
-        tab_hbox.show_all();
+        view_item.connect_events(&edit_view);
+        find_replace.connect_events(&edit_view);
 
-        let pango_ctx = da
-            .get_pango_context()
-            .unwrap_or_else(|| panic!("{}", &gettext("Failed to get Pango context")));
-        let font_list: Vec<String> = pango_ctx
+        edit_view
+    }
+
+    fn get_font_list(pango_ctx: &pango::Context) -> Vec<String> {
+        pango_ctx
             .list_families()
             .iter()
             .filter(|f| f.is_monospace())
             .filter_map(|f| f.get_name())
             .map(|f| f.to_string())
-            .collect();
-        main_state.borrow_mut().fonts = font_list;
+            .collect()
+    }
+
+    fn get_root_box(view_item: &ViewItem, find_replace: &FindReplace) -> Box {
```

Comment:
Hmm...not quite sure if I like the `get_` prefix for this since this isn't something that should be run more than once. Maybe `create_` or `setup_` would be better?